src/test/ui/proc-macro/nonterminal-token-hygiene.rs

```diff
@@ -0,0 +1,27 @@
+// Make sure that marks from declarative macros are applied to tokens in nonterminal.
+
+// check-pass
+// aux-build:test-macros.rs
+
+#![feature(decl_macro)]
+
+#[macro_use]
+extern crate test_macros;
+
+macro_rules! outer {
+    ($item:item) => {
+        macro inner() {
+            recollect! { $item }
```

Comment:
Could you change this to use `print_bang`, and make this test file like https://github.com/rust-lang/rust/blob/0fb00251a578018430bad6687ff74e14031d2a07/src/test/ui/proc-macro/meta-macro-hygiene.rs (e.g. `-Z unpretty=expanded,hygiene` and other flags)? This behavior is quite subtle, and I think it would be good to ensure it never accidentally changes.